UI/MilitaryPanel.cpp

```diff
@@ -41,16 +41,27 @@ MilitaryPanel::MilitaryPanel(GG::X w, int planet_id) :
 
     GG::Connect(m_expand_button->LeftClickedSignal, &MilitaryPanel::ExpandCollapseButtonPressed, this);
 
+    const auto obj = GetUniverseObject(m_planet_id);
+    if (!obj) {
+        ErrorLogger() << "Invalid object id " << m_planet_id;
+        return;
+    }
+
     // small meter indicators - for use when panel is collapsed
-    m_meter_stats.push_back({METER_SHIELD, new StatisticIcon(ClientUI::MeterIcon(METER_SHIELD), 0, 3, false,
+    m_meter_stats.push_back({METER_SHIELD, new StatisticIcon(ClientUI::MeterIcon(METER_SHIELD),
```

Comment:
The values that the icon was flickering between were the initial incorrect default value and the correct value.  These lines initialize the icons with the correct value.